src/query/atomWithInfiniteQuery.ts

```diff
@@ -114,7 +128,7 @@ export function atomWithInfiniteQuery<
 
       const observer = new InfiniteQueryObserver(queryClient, defaultedOptions)
 
-      if (!initialData) {
+      if (typeof initialData === 'undefined') {
```

Comment:
Would you help me understand the background of this change? If this change is necessary, I may misunderstand something and feel checking `=== 'undefined'` might not be enough.

Reply:
`initialData` could be of `null` or `0` type, which would be a defined value, but the previous check would have assumed it was equal to undefined

Reply:
> `initialData` could be of `null` or `0` type

OK, that makes sense now. I wasn't aware of it. But then, a user can do `initialData = undefined`, right? I would introduce a new variable without relying of `undefined`.

Reply:
>  a user can do initialData = undefined, right? I would introduce a new variable without relying of undefined.

Can you give an example, or rephrase? I think if a user were to pass `initialData: undefined` it would be equal to not passing it.

Reply:
Hm, if a user knows `undefined` to lead to Suspense, I think it's fine as is.

I thought about a function:
```js
// case A
options.initialData = () => {
  let something = null
  if (...) {
    something = ...
  }
  return something
}

// case B
options.initialData = () => {
  let something
  if (...) {
    something = ...
  }
  return something
}
```

I expect case A and case B to behave the same.

Reply:
yeah, I think the argument for `null == undefined` is valid. There are cases in which it could be that someone has fetched data on the server where the result is `0` and it's a valid value, but if we just did `!initialData` it would suspend (where the user would expect it to not).

Additionally, if we did support `null` as a valid value for initialData, it would be a way for folks to get around suspense by default for query atoms. It is more of a side effect/hack and I can see why it might not be desirable.

Reply:
okay, my concern was too subtle. I don't mean to do `!initialData`.
Let's just make it 
```js
      if (initialData === undefined) {
```
to be consistent with other code. (We don't assume the `undefined` global variable to be overwritten.)

Reply:
fixed!